lib/Gearman/Spawner/Server.pm

```diff
@@ -32,48 +33,45 @@ sub create {
 # start up a gearmand that exits when its parent process does. returns the
 # address of the listening server and its pid
 sub fork_gearmand {
-    my $class = shift;
-
-    # NB: this relies on Gearman::Spawner::Process allowing use of fork,
+	my $class = shift;
+    
+	# NB: this relies on Gearman::Spawner::Process allowing use of fork,
     # run_periodically, exit_with_parent, and loop as class methods instead of
     # object methods
 
-    # get an unused port from the OS
```

Comment:
You check unused port in parent -> then fork -> try to use previously seen open port in child. So there is time window for port to get reserved by other process. This was seen on our production systems and our Gearman::Spawner::Server process failed to start occasionally.